zebra-chain/src/amount.rs

```diff
@@ -191,6 +191,54 @@ where
     }
 }
 
+impl<C> std::ops::Mul<i64> for Amount<C>
+where
+    C: Constraint,
+{
+    type Output = Result<Amount<C>>;
+
+    fn mul(self, rhs: i64) -> Self::Output {
+        let value = self.0 * rhs;
```

Comment:
Let's handle overflow as an error:

```suggestion
        let value = self.0.checked_mul(rhs).ok_or("overflow in multiplication".into())?;
```

Reply:
Done as part of https://github.com/ZcashFoundation/zebra/pull/1051/commits/364f70eea3f19be9c35d884d861b5096018aa4fd